teps/0089-nonfalsifiable-provenance-support.md

```diff
@@ -59,26 +63,41 @@ There are a couple issues with this:
 2. Tekton Pipelines can't verify that the results it reads weren't modified
 
 This is where SPIRE comes in!
-SPIRE can be used to request signatures and certificates (SVIDs) for a given workload.
-This workload can be anything, including a Result, a Pod spec, or an entire TaskRun yaml.
+SPIRE can be used to request JWTs and certificates (SVIDs) for a given workload (pod for kubernetes).
 
 We're going to use SPIRE to mitigate both of the issues mentioned above.
 
 ## SPIRE Concepts
 Before discussing how SPIRE is going to fix these issues, here's a very basic overview of how it works:
 
-Pods running in the cluster can interact with SPIRE.
-A pod will send some `payload` to SPIRE for signing.
-The SPIRE socket will confirm that the request came from an approved workload (in our case, the Tekton controller or Pods created by the controller) and then it will sign the `payload`.
-SPIRE will return:
-1. The `signature` over the `payload`
-1. A SPIFFE Verifiable Identity Document, or [SVID](https://spiffe.io/docs/latest/spiffe-about/spiffe-concepts/#spiffe-verifiable-identity-document-svid), an X509 certificate containing the public key used to verify the `signature` against the `payload`
+A SPIRE deployment relies on two key components, the SPIRE server, and its associated SPIRE agents.
+- The SPIRE server serves as a central management system for SPIRE, responsible for interfacing with any key material, authorities, and databases. It is also responsible for attesting the agents that are part of its trust domain.
+- The SPIRE server acts as local endpoints for each compute node that is part of the SPIRE deployment.
 
-SPIRE can be used to request certificates (SVIDs) for a given k8s workload and can use these certificates to create signatures for the data we need to verify, i.e. an entire TaskRun yaml and results. For our use case, both the Tekton entrypointer image (running in Pods) and the Tekton Controller will interact with SPIRE.
+Pods running in a cluster can interact with SPIRE through SPIFFE api via the local SPIRE agent socket.
+
+Getting pod identity
+- A pod can request for an identity by talking to the local agent socket. The SPIRE agent attests the identity of the pod,
+and request for an identity for the pod to the SPIRE server.
+- The workload then receives one or more SPIFFE Verifiable Identity Document, or [SVID](https://spiffe.io/docs/latest/spiffe-about/spiffe-concepts/#spiffe-verifiable-identity-document-svid), an X509 certificate together with the associated private key for the pod's identity. The x509 certificate is signed by the SPIRE server's certificate authority.
```

Comment:
im a bit surprised that SPIRE sends the private key to the caller - is there anything that limits how this private key is used after it has been fetched?

tekton tasks are executed as pods, which in this proposal would have the SPIRE socket mounted into them - it sounds like a user could add steps into their Task (and resulting pod) that could fetch this private key and upload it somewhere else for use later - am i understanding correctly?

(im guessing im missing a piece of the puzzle that prevents this from happening)

Reply:
Okay thanks for the extra details @lumjjb ! I need a bit of time to chew on this and understand the implications but no reason to block these changes in the meantime!